text/3660-crates-io-crate-deletions.md

```diff
@@ -0,0 +1,123 @@
+- Start Date: 2024-06-20
+- RFC PR: [rust-lang/rfcs#3660](https://github.com/rust-lang/rfcs/pull/3660)
+
+# Summary
+
+This RFC proposes a mechanism for crate authors to delete their crates from crates.io under certain conditions.
+
+
+# Motivation
+
+There are a variety of reasons why a crate author might want to delete a crate or version from crates.io:
+
+* You published something accidentally.
+* You wanted to test crates.io.
+* You published content you didn't intend to be public.
+* You want to rename a crate. (The only way to rename a package is to re-publish it under a new name)
+
+The current [crates.io usage policy](https://crates.io/policies) says:
+
+> Crate deletion by their owners is not possible to keep the registry as immutable as possible.
+
+This restriction makes sense for the majority of crates that have been around for a while and are actively used, but the above list of reasons shows that there are valid use cases for allowing crate authors to delete their crates without having to contact the crates.io team.
+
+To make this process easier for our users and to reduce the workload of the crates.io team dealing with such support requests, we propose to codify our current set of informal rules into a formal policy that allows crate authors to delete their crates themselves under certain conditions (see below).
+
+
+# Proposal
+
+We propose to allow crate authors to delete their **crates** from crates.io under the following conditions:
+
+* The crate has been published for less than 72 hours,
+* or if all the following conditions are met:
+  * The crate has a single owner,
```

Comment:
This doesn't limit much. It's easy for one owner to kick all other owners.

Reply:
that is a fair point. eventually I'd like to improve our owner system to have different access levels, but you are correct that for now this is technically not a hard limitation. it does at least add a bit of friction though, so that crates are less likely to be deleted unintentionally (see comments above).

Reply:
For better or worse, teams at least offers some "owner control".  iirc team members can't change permissions.